From 99ecedc0930a99fb4d084208964e4c7922a6efec Mon Sep 17 00:00:00 2001
From: Masanori Kakura <kakurasan@gmail.com>
Date: Tue, 25 Sep 2018 18:04:06 +0900
Subject: [PATCH] environment: Use os.path.basename() when checking
 mesonlib.meson_command

Without this, when one of Meson's parent directories contains
"python", unexpected option "-u" will be passed to Meson.
---
 mesonbuild/environment.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/mesonbuild/environment.py b/mesonbuild/environment.py
index 7a44f2fd..a0c2fda4 100644
--- a/mesonbuild/environment.py
+++ b/mesonbuild/environment.py
@@ -375,7 +375,7 @@ class Environment:
 
     def get_build_command(self, unbuffered=False):
         cmd = mesonlib.meson_command[:]
-        if unbuffered and 'python' in cmd[0]:
+        if unbuffered and 'python' in os.path.basename(cmd[0]):
             cmd.insert(1, '-u')
         return cmd
 
-- 
2.19.0

